lapack/gonum/dgetc2.go

```diff
@@ -0,0 +1,109 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gonum
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/blas/blas64"
+)
+
+// Dgetc2 computes an LU factorization with complete pivoting of the
+// n-by-n matrix A. The factorization has the form
+//  A = P * L * U * Q,
+// where P and Q are permutation matrices, L is lower triangular with
+// unit diagonal elements and U is upper triangular.
+//
+// Outputs are A, ipiv, jpiv and k. k is non-negative if U(k, k)
+// is likely to produce overflow when  e try to solve for x in Ax = b.
+// So U is perturbed to avoid the overflow.
```

Comment:
This doesn't look like our normal style of docs. I'll let @vladimir-ch comment on this.

Reply:
I've looked at `Dgelq2` to make the documentation better for last commit.

Reply:
It's ok. I'll polish it after it's merged.